tooling/noir_js_types/src/types.ts

```diff
@@ -45,7 +45,9 @@ export interface Backend {
  * */
 export type ProofData = {
   /** @description Public inputs of a proof */
-  publicInputs: Uint8Array;
+  publicInputs: string[];
+  /** @description The public inputs as a witness map */
+  publicInputsMap: WitnessMap;
```

Comment:
I'm a wary of this as a malicious user could post different data in the two fields.

Reply:
Could we not add an optional aggregation object field to this object? We'd need a small helper function when passing this into the smart contract but would otherwise give the best of both worlds afaict.

Reply:
That's true... we could always have the publicInputs vector and provide the publicInputsToWitnessMap function though

Reply:
hmm, yeah I'd be fine with that. Let's remove the `publicInputsMap` field from this however.